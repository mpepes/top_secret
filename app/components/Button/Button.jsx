import React from 'react';
import PropsTypes from 'prop-types';

const Button = ({
    onClick,
    children,
    disabled,
    type,
    className,
}) => (
    <button
        onClick={onClick}
        disabled={disabled}
        type={type}
        className={className}
    >
        {children}
    </button>
);

Button.propTypes = {
    onClick: PropsTypes.func.isRequired,
    children: PropsTypes.oneOfType([
        PropsTypes.string,
        PropsTypes.node,
    ]).isRequired,
    disabled: PropsTypes.bool,
    type: PropsTypes.string,
    className: PropsTypes.string,
};

Button.defaultProps = {
    disabled: false,
    type: 'button',
    className: '',
};

export default Button;
