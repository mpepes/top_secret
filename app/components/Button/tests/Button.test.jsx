import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import { shallow, mount } from 'enzyme';

import Button from '../Button';

const renderer = new ShallowRenderer();

const props = {
    onClick: jest.fn(),
};

describe('SearchInput', () => {
    describe('Snapshot', () => {
        it('should render SearchInput with no value', () => {
            const wrapper = <Button {...props}>Test</Button>;
            const button = renderer.render(wrapper);
            expect(button).toMatchSnapshot();
        });
    });

    describe('Unit', () => {
        beforeEach(() => {
            jest.clearAllMocks();
        });

        it('should call onClick when clicked', () => {
            const button = shallow(<Button {...props}>Test</Button>);
            button.find('button').simulate('click');
            expect(props.onClick).toHaveBeenCalledTimes(1);
        });

        it('should not call onClick when button is disabled', () => {
            props.disabled = true;
            const button = mount(<Button {...props}>Test</Button>);
            button.find('button').simulate('click');
            expect(props.onClick).toHaveBeenCalledTimes(0);
        });
    });
});
