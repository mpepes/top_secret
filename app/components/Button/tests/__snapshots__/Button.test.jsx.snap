// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchInput Snapshot should render SearchInput with no value 1`] = `
<button
  className=""
  disabled={false}
  onClick={[MockFunction]}
  type="button"
>
  Test
</button>
`;
